repose-aggregator/commons/commons-utilities/src/main/scala/org/openrepose/commons/utils/servlet/http/HttpServletResponseWrapper.scala

```diff
@@ -99,6 +102,48 @@ class HttpServletResponseWrapper(originalResponse: HttpServletResponse, headerMo
   def this(originalResponse: HttpServletResponse, headerMode: ResponseMode, bodyMode: ResponseMode) =
     this(originalResponse, headerMode, bodyMode, originalResponse.getOutputStream)
 
+  override def setStatus(i: Int, s: String): Unit = {
+    message = s
+    super.setStatus(i, s)
+  }
+
+  // todo: how does this affect the container processing invoked by sendError? should we track that this method was
+  // called? should commitToResponse take a parameter indicating if an error should bubble up?
+  /** Sets a status code and message on the first line of the HTTP response. The container may do additional processing
+    * to, for example, generate an error page as the response entity.
+    *
+    * Note that in any mutable mode, this method will call through to the [[sendError(i)]] method of the wrapped
+    * response. As a result, the container will not have an opportunity to perform additional processing.
+    *
+    * @param i the status code to set
+    */
+  override def sendError(i: Int): Unit = {
+    committed = true
+    if (headerMode == ResponseMode.MUTABLE || bodyMode == ResponseMode.MUTABLE) {
+      super.setStatus(i)
+    } else {
+      super.sendError(i)
+    }
+  }
+
+  /** See [[sendError(i)]].
+    *
+    * @param i the status code to set
+    */
+  override def sendError(i: Int, s: String): Unit = {
+    committed = true
+    message = s
+    if (headerMode == ResponseMode.MUTABLE || bodyMode == ResponseMode.MUTABLE) {
+      super.setStatus(i)
```

Comment:
Yes. In fact, we never need to chain through to sendError since we wrap the request in the PowerFilter with a mode set to mutable.

Also, we can probably flush here as well.